File: src/main/kotlin/binaryTreeLevelOrderTraversal/BinaryTreeLevelOrderTraversal.kt
package binaryTreeLevelOrderTraversal

import utils.TreeNode
import java.util.LinkedList

class BinaryTreeLevelOrderTraversal {
    fun levelOrder(root: TreeNode?): List<List<Int>> {
        if (root == null) return emptyList()
        val result = mutableListOf<List<Int>>()
        val queue = LinkedList<TreeNode>()
        queue.add(root)
        while (queue.isNotEmpty()) {
            result.add(queue.toList().map { it.`val` })
            for (i in queue.indices) {
                val cur = queue.poll()
                cur.left?.let { queue.add(it) }
                cur.right?.let { queue.add(it) }
            }
        }
        return result
    }
}

File: src/main/python/binaryTreeLevelOrderTraversal/BinaryTreeLevelOrderTraversal.py
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> list[list[int]]:
        result = []
        if root is None:
            return result
        queue = [root]
        while queue:
            size = len(queue)
            tmp = []
            while size:
                size -= 1
                cur = queue.pop(0)
                tmp.append(cur.val)
                if cur.left is not None:
                    queue.append(cur.left)
                if cur.right is not None:
                    queue.append(cur.right)
            result.append(tmp)
        return result
